worker/uniter/uniter.go

```diff
@@ -92,28 +98,36 @@ type Uniter struct {
 	updateStatusAt TimedSignal
 }
 
+// UniterParams hold all the necessary parameters for a new Uniter.
+type UniterParams struct {
+	St                 *uniter.State
+	UnitTag            names.UnitTag
+	LeadershipManager  coreleadership.LeadershipManager
+	DataDir            string
+	HookLock           *fslock.Lock
+	MetricsTimer       *timerChooser
+	UpdateStatusSignal TimedSignal
+}
+
 // NewUniter creates a new Uniter which will install, run, and upgrade
 // a charm on behalf of the unit with the given unitTag, by executing
 // hooks and operations provoked by changes in st.
-func NewUniter(
-	st *uniter.State,
-	unitTag names.UnitTag,
-	leadershipManager coreleadership.LeadershipManager,
-	dataDir string,
-	hookLock *fslock.Lock,
-) *Uniter {
+func NewUniter(uniterParams *UniterParams) *Uniter {
 	u := &Uniter{
-		st:                st,
-		paths:             NewPaths(dataDir, unitTag),
-		hookLock:          hookLock,
-		leadershipManager: leadershipManager,
-		collectMetricsAt:  inactiveMetricsTimer,
-		updateStatusAt:    updateStatusSignal,
+		st:                uniterParams.St,
+		paths:             NewPaths(uniterParams.DataDir, uniterParams.UnitTag),
+		hookLock:          uniterParams.HookLock,
+		leadershipManager: uniterParams.LeadershipManager,
+		metricsTimer:      uniterParams.MetricsTimer,
+		// TODO(perrito666) make these signals directly fetched from
+		// the timer.
+		collectMetricsAt: uniterParams.MetricsTimer.defaultTimer(),
```

Comment:
why defaultTimer and not inactiveTimer? we're making an explicit choice to use an *inactive* timer at this point, not because it's default but because it's not safe to read the data that would allow us to choose otherwise; and later we will make an explicit choice between active and inactive timers depending on the charm's features. There's nothing default about inactivity.